rosetta-source/src/main/rosetta/event-position-type.rosetta

```diff
@@ -38,6 +44,49 @@ type AggregationParameters: <" Parameters to be used to filter events that are r
 
 type Portfolio: <" A Portfolio represents an aggregation of multiple Positions, by describing the parameters that this Portfolio should be aggregated based on. The resulting PortfolioState is calculated using these aggregation parameters as inputs, by aggregating all the Events that are relevant to this Portfolio. The concept of Portfolio works at all levels in the model: from the highest for a given LegalEntity for instance, to the lowest to account for security substitutions in a secutity financing transaction. As such, Portfolio can be used either above or below the Contract level.">
+	  aggregationParameters AggregationParameters (1..1) <"Describes the portfolio by describing how to aggregate all its relevant Events.">
+	  portfolioState PortfolioState (1..1) <"Describes the state of the Portfolio as a list of Positions resulting from the aggregation.">
+
+type Inventory: <"A data type that can be used to describe an inventory of securities.">
+    inventoryRecord InventoryRecord (0..*) <"An array holding the list of inventory being described. Each element in the inventoryRecord array represents an individual piece of inventory i.e. a security.">
+
+type InventoryRecord: <"An individual piece of inventory. This represents a single security.">
+    identifer AssignedIdentifier (1..1) <"Unique identifier for this record. This can be used to uniquely identify a specific piece of inventory.">
+    security Security (1..1) <"The security details.">
+
+type AvailableInventory: <"A data type that can be used to describe the inventory of securities that a party holds. The securities are held in the AvailableInventoryRecord, with each item in the array being an individual security and its associated criteria. Criteria can include the quantity available, the rate at which the security is available to borrow at, as well as other details that can affect the decision as to whether a party wants to utilise the securities listed.">
+    messageInformation MessageInformation (0..1) <"Allows details related to the availability messaging use case to be defined"> 
+    party Party (0..*) <"Defines all parties involved for the list of inventory records in this set of inventory. For example, when used to describe securities lending availability, this could hold the sender of the availability, the intended recipient, the beneficial owner(s), the lender (which may differ from the sender as the lender may have the same piece of availability going through multiple agents), an agent or a venue.">
+    partyRole PartyRole (0..*) <"Defines the role(s) that party(ies) may have in relation to the inventory.">
+    availableInventoryRecord AvailableInventoryRecord (0..*) <"An array holding the list of inventory being described. Each element in the inventoryRecord array represents an individual piece of inventory i.e. a security.">
+	
+    condition ValidPartyRole: <"Restrict roles to be only those that would be associated to the overall list of availability">
```

Comment:
Don't you also want to test that if, say, `Lender` is specified at the `AvailableInventory` level, then `Lender` is absent from `partyRole` at the `AvailableInventoryRecord`? (to avoid the risk of inconsistency)

Reply:
See also other comment below re the semantics of the condition

Reply:
I did raise this at the ISLA CDM WG last Thursday but the consensus was that there could be occasions where the roles were not mutually exclusive. It is also true to say that a party may fulfil multiple roles (thus ```partyRole``` at the record level is 0 to many) which again could lead to the same role being required at the top level and the individual record level.